.ci/ci_cache_components.sh

```diff
@@ -0,0 +1,179 @@
+#!/bin/bash
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+cidir=$(dirname "$0")
+source "${cidir}/../lib/common.bash"
+source "${cidir}/lib.sh"
+WORKSPACE=${WORKSPACE:-$(pwd)}
+kata_dir="/usr/share/kata-containers"
+
+cache_qemu_artifacts() {
+	current_qemu_version=$(get_version "assets.hypervisor.qemu.version")
+	qemu_tar="kata-qemu-static.tar.gz"
+	build_cache "$qemu_tar" "${current_qemu_version}"
+}
+
+cache_kernel_artifacts() {
+	for k in "${kata_dir}"/*.container; do
+		echo "Adding ${k}"
+		real_path=$(readlink -f "${k}")
+		#Get version from binary (format: vmlinu{z|x}-4.19.24-25)
+		kernel_binary=$(basename "${real_path}")
+		current_kernel_version=$(echo "${kernel_binary}" | cut -d- -f2-)
+		build_cache "${k}" "${current_kernel_version}"
+	done
+}
+
+cache_image_artifacts() {
+	image="${kata_dir}/kata-containers.img"
+	path=$(readlink -f "${image}")
+	image_version=$(echo $(basename "${path}"))
+	build_cache "${image}" "${image_version}"
+	create_image_tar "${image}"
+}
+
+cache_nemu_artifacts() {
+	nemu_version=$(get_version "assets.hypervisor.nemu.version")
+	nemu_tar="kata-nemu-static.tar.gz"
+	build_cache "${nemu_tar}" "${nemu_version}"
+}
+
+cache_image_initrd_artifacts() {
+	image="${kata_dir}/kata-containers-initrd.img"
+	path=$(readlink -f "${image}")
+	image_version=$(echo $(basename "${path}"))
+	build_cache "${image}" "${image_version}"
+	create_image_tar "${image}"
+}
+
+create_image_tar(){
+	image="$1"
+	sha256sum -c "sha256sum-${image}"
+	tar -cJf "$(cat latest-${image}).targ.xz" "$(cat latest-${image})"
+	sha256sum "$(cat latest-${image}).tar.xz" > "sha256sum-${image}-tarball"
+	sha256sum -c "sha256sum-${image}-tarball"
+}
+
+build_cache() {
+	component_name="$1"
+	component_version="$2"
+	check_image=$(echo "${component_name}" | grep image)
+	check_nemu=$(echo "${component_name}" | grep nemu)
+	if [ ! -z "${check_image}" ]; then
+		check_initrd=$(echo "${component_name}" | grep initrd)
+		if [ ! -z "${check_initrd}" ]; then
+			image="${kata_dir}/kata-containers-initrd.img"
+		else
+			image="${kata_dir}/kata-containers.img"
+		fi
+		path=$(readlink -f "${image}")
+		echo $(basename "${path}") > "latest-${image}"
+		sudo cp "${path}" "${kata_dir}/osbuilder-${image}.yaml"
+	else
+		echo "${component_version}" >  "latest"
+	fi
+
+	if [ ! -z "${check_nemu}" ]; then
+		cp -a "/tmp/${component_name}" .
+	fi
+
+	sudo chown -R "${USER}:${USER}" .
+	sha256sum "${component_name}" > "sha256sum-${component_name}"
+	cat "sha256sum-${component_name}"
+}
+
+help() {
+echo "$(cat << EOF
+Usage: $0 "[options]"
+	Description:
+		This script builds the cache of several kata components.
+	Options:
+		-a      Run kernel cache
+		-b      Run image cache
+		-c      Run nemu cache
+		-d      Run image initrd cache
+		-e      Run qemu cache
+		-h      Shows help
+EOF
+)"
+}
+
+main() {
+	local OPTIND
+	while getopts ":abcdeh:" opt
```

Comment:
Not necessary, but you could use `getopt(1)` and the tricks employed by the static check script to dynamically build the `usage()` function output:

- https://github.com/kata-containers/tests/blob/master/.ci/static-checks.sh#L54
- https://github.com/kata-containers/tests/blob/master/.ci/static-checks.sh#L85..L116
- https://github.com/kata-containers/tests/blob/master/.ci/static-checks.sh#L996..L1001

That way, `usage()` is guaranteed to be correct even if you change options at a later date.